modules/caddytls/leaffileloader.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2015 Matthew Holt and The Caddy Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package caddytls
+
+import (
+	"crypto/x509"
+	"fmt"
+	"os"
+
+	"github.com/caddyserver/caddy/v2"
+)
+
+func init() {
+	caddy.RegisterModule(LeafFileLoader{})
+}
+
+// LeafFileLoader loads leaf certificates from disk.
+type LeafFileLoader []LeafCertFile
+
+// Provision implements caddy.Provisioner.
+func (fl LeafFileLoader) Provision(ctx caddy.Context) error {
+	repl, ok := ctx.Value(caddy.ReplacerCtxKey).(*caddy.Replacer)
+	if !ok {
+		repl = caddy.NewReplacer()
+	}
+	for k, pair := range fl {
+		for i, tag := range pair.Tags {
+			pair.Tags[i] = repl.ReplaceKnown(tag, "")
+		}
+		fl[k] = LeafCertFile{
+			LeafCertificate: repl.ReplaceKnown(pair.LeafCertificate, ""),
+			Format:          repl.ReplaceKnown(pair.Format, ""),
+			Tags:            pair.Tags,
+		}
+	}
+	return nil
+}
+
+// CaddyModule returns the Caddy module information.
+func (LeafFileLoader) CaddyModule() caddy.ModuleInfo {
+	return caddy.ModuleInfo{
+		ID:  "leaf_cert_loader.load_files",
```

Comment:
The namespace is `tls.leaf_cert_loader`, so the implementation module should have the namespace as the prefix, not just the middle part. In other words, the namespace here should be `tls.leaf_cert_loader.load_files`, though I think the module name `files` (the suffix part) is more apt because we already know it's a loader per the namespace.

Reply:
I was planning on adding the `tls` prefix but I completely forgot to do it. I agree with the naming though. I just updated the code to include those two changes for the four modules I added. Thank you for your feedback!